src/main/java/org/dita/dost/module/CleanPreprocessModule.java

```diff
@@ -263,11 +263,12 @@ URI getCommonBase(final URI left, final URI right) {
                 //
             } else {
                 final int common = Math.max(0, i);
-                final String path = Arrays.asList(la)
-                        .subList(0, common)
-                        .stream()
-                        .collect(Collectors.joining("/")) + "/";
-                return URLUtils.setPath(left, path);
+                final List<String> commons = Arrays.asList(la).subList(0, common);
+                if (commons != null && commons.size() > 1) {
```

Comment:
With a test

```java
assertEquals(create("file:/foo/"), module.getCommonBase(create("file:/foo/a/b/c"), create("file:/bar/b/c/d")));
```

this code passes, but the result is incorrect. The correct results should be `file:/`. The code makes assumption that the first directory in the path token should be retained and the resulting path can never start from root. This only applies to Windows, not *NIX.

Reply:
The problem is in fact that if the common path is only `file:/` a NPE appears in the HTML5 transformation.
And you're right this problem is only Windows related as Linux doesn't have drives.